daemon/src/wallet.rs

```diff
@@ -134,16 +138,13 @@ impl Actor {
 
 #[xtra_productivity]
 impl Actor {
-    pub fn handle_reinitialise(
-        &mut self,
-        msg: Reinitialise,
-        ctx: &mut xtra::Context<Self>,
-    ) -> Result<()> {
+    pub fn handle_restore(&mut self, msg: Restore, ctx: &mut xtra::Context<Self>) -> Result<()> {
         let client = bdk::electrum_client::Client::new(&self.electrum_rpc_url)
             .context("Failed to initialize Electrum RPC client")?;
 
-        let ext_priv_key =
-            ExtendedPrivKey::new_master(self.wallet.network(), msg.seed_words.as_bytes())?;
+        let seed = Seed::try_from(msg.mnemonic)?;
+
+        let ext_priv_key = seed.derive_extended_priv_key(self.wallet.network())?;
 
         let db = bdk::database::MemoryDatabase::new();
```

Comment:
We need to store the new seed words on disk here, otherwise they are completely lost!

Reply:
I moved saving to disk to the `TakerActorSystem` where `Restore` is sent. This means the `wallet::Actor` does not need to know about the path of the wallet seed file.

Reply:
Ah right, that makes sense yes. That will make it also easier to solve this via restart the wallet actor if we ever want to.